src/docfx/lib/process/InterProcessReaderWriterLock.cs

```diff
@@ -76,32 +77,35 @@ private static FileStream WaitFile(string name, string path, FileAccess access,
 
         public void Dispose()
         {
-            if (_fileStream != null)
+            using (new GlobalMutex(_lockHash))
             {
-                using (new GlobalMutex(_filePath))
-                {
-                    _fileStream.Dispose();
-                }
+                _fileStream.Dispose();
             }
         }
 
-        private struct GlobalMutex : IDisposable
+        private readonly struct GlobalMutex : IDisposable
         {
             private readonly Mutex _mutex;
 
-            public GlobalMutex(string name)
+            public GlobalMutex(string hash)
             {
-                _mutex = new Mutex(initiallyOwned: false, $"Global\\{HashUtility.GetMd5Hash(name)}");
-                _mutex.WaitOne();
+                _mutex = new Mutex(initiallyOwned: false, $"Global\\{hash}");
+
+                try
+                {
+                    _mutex.WaitOne();
+                }
+                catch (AbandonedMutexException)
+                {
+                    // When another process/thread exited without relasing its mutex,
+                    // this exception is thrown and we've successfully acquired the mutex.
+                }
             }
 
             public void Dispose()
             {
-                if (_mutex != null)
-                {
-                    _mutex.ReleaseMutex();
-                    _mutex.Dispose();
-                }
+                _mutex.ReleaseMutex();
```

Comment:
what if it throws during instancing a new `Mutex`?

Reply:
AbandonedMutex exception only throws in `WaitOne` method